addons/zylann.scatter/tools/palette.gd

```diff
@@ -101,7 +101,9 @@ func _on_AddButton_pressed():
 
 
 func _on_RemoveButton_pressed():
-	for item in _item_list.get_selected_items():
+	var selected = _item_list.get_selected_items()
+	selected.invert()
+	for item in selected:
```

Comment:
I think a better idea would be actually to emit the list of all patterns that got removed, and process it accordingly in the plugin. Another reason to do it this way would be to make the whole action undoable (otherwise it would undo only one by one)

Reply:
Sounds reasonable, done.